### PR TITLE
CAL-61: Added the ability to specify sources by name on the NSILI Endpoint

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Added the ability to specify sources by name on the NSILI Endpoint for restricting enterprise searches
 - Added query params to the source that get applied to each outgoing query to that source
 - Fixed an issue with setting the port for the endpoint was causing port conflicts with the source
 - Resolved 406 issues when attempting self-federation
 - Resolved self-federation federated searching problems Self NSILI Source - NSILI Endpoint - External NSIL implementation

#### Who is reviewing it?
@jaymcnallie @bdeining @millerw8 @kcwire 

#### How should this be tested?
Run the JUnit tests

#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-61

#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java [hunk 1]
```diff
@@ -72,19 +74,21 @@
...
+    private List<String> querySources = new ArrayList<>();
+
     private static final Logger LOGGER = LoggerFactory.getLogger(NsiliEndpoint.class);
 
     public NsiliEndpoint() {
         LOGGER.debug("NSILI Endpoint constructed");
     }
 
     public void shutdown() {
+        LOGGER.error("Attempting to Shutdown the ORB");
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java]: Should this be error level?

Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java]: No. Will change that to INFO.

**Comment by bdeining** [comment]
:+1: after rebase

**Comment by troymohl** [comment]
I have rebased this and addressed @bdeining  comment.

**Comment by jaymcnallie** [comment]
LGTM 👍

**Pull request merged**